### PR TITLE
Fix for MEMKIND_HBW_ALL

### PR DESCRIPTION
**Pull request by michalbiesek**
Based on #493. I recommend reviewing this commit by commit (first commit is done in #493).

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/494)

### CODE REVIEW

File: include/memkind/internal/memkind_bitmask.h [hunk 1]
```diff
@@ -12,14 +12,12 @@ extern "C" {
...
+                         memkind_node_variant_t node_variant);
 void set_bitmask_for_all_closest_numanodes(unsigned long *nodemask,
-                                           unsigned long maxnode, const void *closest_numanode,
-                                           int num_cpu);
+                                           unsigned long maxnode, const void *numanode);
 int set_bitmask_for_current_closest_numanode(unsigned long *nodemask,
-                                             unsigned long maxnode, const void *closest_numanode,
-                                             int num_cpu);
+                                             unsigned long maxnode, const void *numanode);
 int memkind_env_get_nodemask(char *nodes_env, struct bitmask **bm);
```
Comment (PatKamin) [inline include/memkind/internal/memkind_bitmask.h:21]: nit: It's not part of this commit, but you can remove this extra line.

File: include/memkind/internal/memkind_mem_attributes.h [hunk 2]
```diff
@@ -17,7 +17,7 @@ typedef enum memory_attribute_t {
 
 int get_per_cpu_local_nodes_mask(struct bitmask ***nodes_mask,
                                  memkind_node_variant_t node_variant, memory_attribute_t attr);
-int set_closest_numanode_mem_attr(void **closest_numanode, int num_cpu,
+int set_closest_numanode_mem_attr(void **numanode,
                                   memkind_node_variant_t node_variant);
```
Comment (PatKamin) [inline include/memkind/internal/memkind_mem_attributes.h:21]: nit: Possibly can be in a single line.

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 8 of 8 files at r1, 7 of 7 files at r2.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/494)* status: all files reviewed, 4 unresolved discussions (waiting on @michalbiesek)

___
*[include/memkind/internal/memkind_private.h, line 103 at r2](https://reviewable.io/reviews/memkind/memkind/494#-MS1u1m1AyJzz5aWf_s0:-MS1u1m2CjxkigBCu8W3:b-z8tcpv) ([raw file](https://github.com/memkind/memkind/blob/499d5d1e75dfca010e092fa29a642da80da4539c/include/memkind/internal/memkind_private.h#L103)):*
> ```objc
>     NODE_VARIANT_SINGLE = 1,
>     NODE_VARIANT_MAX = 2,
>     NODE_VARIANT_ALL = 2,
> ```

MAX == ALL?
___
*[src/memkind-hbw-nodes.c, line 53 at r2](https://reviewable.io/reviews/memkind/memkind/494#-MS1uISm0X1_lJwqL5a0:-MS1uISm0X1_lJwqL5a1:b-27iq5r) ([raw file](https://github.com/memkind/memkind/blob/499d5d1e75dfca010e092fa29a642da80da4539c/src/memkind-hbw-nodes.c#L53)):*
> ```C
>     // ensuring functions in numa library are defined
>     if (numa_available() == -1) {
>         return 3;
> ```

not a part of this commit but comment above says, that ret value is in range 0-2. What is 3?

File: include/memkind/internal/memkind_private.h [hunk 3]
```diff
@@ -99,7 +99,9 @@ struct memkind_config {
 typedef enum memkind_node_variant_t {
     NODE_VARIANT_MULTIPLE = 0,
     NODE_VARIANT_SINGLE = 1,
-    NODE_VARIANT_MAX = 2
+    NODE_VARIANT_MAX = 2,
+    NODE_VARIANT_ALL = 2,
```
Comment (PatKamin) [inline include/memkind/internal/memkind_private.h:103]: Why same value?

**Review by michalbiesek — COMMENTED** [review]
The are two purposes of this enum:
- first handle a logic in functions like memkind_hi_cap_find,get_per_cpu_local_nodes_mask or set_closest_numanode_mem_attr
- second, handle the size of the arrays for storing expected mapping structures see e.g dax_numanode_t

In the case of HBW and DAX, we support 3 variants, and in other cases only 2 variants. The  NODE_VARIANT_MAX_EXT and  NODE_VARIANT_MAX are responsible for this.

> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/494#-:-MS25bGh-MUu1SHGHz35:bf62f00)* status: all files reviewed, 3 unresolved discussions (waiting on @bratpiorka and @PatKamin)

___
*[include/memkind/internal/memkind_bitmask.h, line 23 at r1](https://reviewable.io/reviews/memkind/memkind/494#-MRzPeOfQ6qa81GJhLYu-r1-23:-MS24rwV55cfcS3O6PCI:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/d904787b4c8b84880cf13fc4c626f90f0b5a9dcc/include/memkind/internal/memkind_bitmask.h#L23)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

nit: It's not part of this commit, but you can remove this extra line.
</blockquote></details>

Done.
___
*[include/memkind/internal/memkind_mem_attributes.h, line 21 at r1](https://reviewable.io/reviews/memkind/memkind/494#-MRzPeOfQ6qa81GJhLYv-r1-21:-MS255SH7CWNnt7BI7Qg:b-sk4a3t) ([raw file](https://github.com/memkind/memkind/blob/d904787b4c8b84880cf13fc4c626f90f0b5a9dcc/include/memkind/internal/memkind_mem_attributes.h#L21)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

nit: Possibly can be in a single line.
</blockquote></details>

Astyle forbids it.
___
*[include/memkind/internal/memkind_private.h, line 103 at r2](https://reviewable.io/reviews/memkind/memkind/494#-MS1u1m1AyJzz5aWf_s0:-MS25TJS23OMyqABz1o-:bvibkci) ([raw file](https://github.com/memkind/memkind/blob/499d5d1e75dfca010e092fa29a642da80da4539c/include/memkind/internal/memkind_private.h#L103)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

MAX == ALL?
</blockquote></details>

Done. Changed to ALL = MAX
___
*[src/memkind-hbw-nodes.c, line 53 at r2](https://reviewable.io/reviews/memkind/memkind/494#-MS1uISm0X1_lJwqL5a0:-MS25Zqk2OpHebdJtQ1J:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/499d5d1e75dfca010e092fa29a642da80da4539c/src/memkind-hbw-nodes.c#L53)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

not a part of this commit but comment above says, that ret value is in range 0-2. What is 3?
</blockquote></details>

Done.

**Review by PatKamin — COMMENTED** [review]
:lgtm:

> Reviewed 4 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/494#-:-MS2A4vg1mnEz7BYFfXB:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @bratpiorka)

**Review by bratpiorka — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/494#-:-MS2CSRKDpzROpcZ5JtU:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @michalbiesek)
